packages/gatsby/src/commands/develop.ts

```diff
@@ -1,51 +1,74 @@
-const express = require(`express`)
-const graphqlHTTP = require(`express-graphql`)
-const graphqlPlayground = require(`graphql-playground-middleware-express`)
-  .default
-const graphiqlExplorer = require(`gatsby-graphiql-explorer`)
-const { formatError } = require(`graphql`)
-const got = require(`got`)
-const webpack = require(`webpack`)
-const webpackConfig = require(`../utils/webpack.config`)
-const bootstrap = require(`../bootstrap`)
-const { store, emitter } = require(`../redux`)
-const { syncStaticDir } = require(`../utils/get-static-dir`)
-const buildHTML = require(`./build-html`)
-const { withBasePath } = require(`../utils/path`)
-const report = require(`gatsby-cli/lib/reporter`)
-const launchEditor = require(`react-dev-utils/launchEditor`)
-const formatWebpackMessages = require(`react-dev-utils/formatWebpackMessages`)
-const chalk = require(`chalk`)
-const address = require(`address`)
-const cors = require(`cors`)
-const telemetry = require(`gatsby-telemetry`)
-const WorkerPool = require(`../utils/worker/pool`)
-
-const withResolverContext = require(`../schema/context`)
-const sourceNodes = require(`../utils/source-nodes`)
-const createSchemaCustomization = require(`../utils/create-schema-customization`)
-const websocketManager = require(`../utils/websocket-manager`)
-const getSslCert = require(`../utils/get-ssl-cert`)
-const { slash } = require(`gatsby-core-utils`)
-const { initTracer } = require(`../utils/tracer`)
-const apiRunnerNode = require(`../utils/api-runner-node`)
-const db = require(`../db`)
-const detectPortInUseAndPrompt = require(`../utils/detect-port-in-use-and-prompt`)
-const onExit = require(`signal-exit`)
-const queryUtil = require(`../query`)
-const queryWatcher = require(`../query/query-watcher`)
-const requiresWriter = require(`../bootstrap/requires-writer`)
-const {
+import url from "url"
+import fs from "fs"
+import openurl from "better-opn"
+import chokidar from "chokidar"
+
+import { PackageJson } from "gatsby"
+import glob from "glob"
+import express from "express"
+import got from "got"
+import webpack from "webpack"
+import graphqlHTTP from "express-graphql"
+import graphqlPlayground from "graphql-playground-middleware-express"
+import graphiqlExplorer from "gatsby-graphiql-explorer"
+import { formatError } from "graphql"
+
+import webpackConfig from "../utils/webpack.config"
+import bootstrap from "../bootstrap"
+import { store, emitter } from "../redux"
+import { syncStaticDir } from "../utils/get-static-dir"
+import buildHTML from "./build-html"
+import { withBasePath } from "../utils/path"
+import report from "gatsby-cli/lib/reporter"
+import launchEditor from "react-dev-utils/launchEditor"
+import formatWebpackMessages from "react-dev-utils/formatWebpackMessages"
+import chalk from "chalk"
+import address from "address"
+import cors from "cors"
+import telemetry from "gatsby-telemetry"
+import WorkerPool from "../utils/worker/pool"
+import http from "http"
+import https from "https"
```

Comment:
I wonder what the impact is of depending on this at build time rather than runtime.
Might be benign but it's definitely a change.

Reply:
oof, i'll measure. we only get type checking on modules (not commonjs requires). When i made the joke about shaving off 0.043s it was in reference to this, but didn't even think about the build time cost that this adds to every run.. lol

Reply:
costs `0.012ms` to import the `https` module. So I suppose it's actually a small net win if you are running in ssl, but if you are not we are adding that cost to the import. Seemingly benign for the benefit of types working

Reply:
Probably nothing, cool :)